src/algorithm/mixed_radix.rs

```diff
@@ -93,16 +97,16 @@ impl<T: FFTnum> FFTAlgorithm<T> for MixedRadix<T> {
 /// This struct is the same as MixedRadixSingle, except it's specialized for the case where both inner FFTs are butterflies
 pub struct MixedRadixDoubleButterfly<T> {
     width: usize,
-    width_size_fft: ButterflyEnum<T>,
+    width_size_fft: Rc<FFTButterfly<T>>,
```

Comment:
Ah, no that makes sense. I think `Rc` is fine here, I was just curious because I didn't see the caching.